File: Lexer/RightParenToken.cpp
#include "token.h"
#include <string>
using namespace std;

class RightParen : public Token {
	bool Token::equals()
	{

	}

	int Token::hashCode()
	{
		return 4;
	}

	string Token::toString()
	{
		return ")";
	}
};

File: src/main/java/parser/RightParenToken.java
package parser;

public class RightParenToken implements Token{
	public boolean equals(final Object other) {
		return other instanceof RightParenToken;
	}
	
	public int hashCode() {
		return 10;
	}
	
	public String toString() {
		return "RightParenToken";
	}

}
